tests/pytests/test_search_params.py

```diff
@@ -414,3 +414,164 @@ def test_fuzzy(env):
+    # TEXT field should return the original value.
+    
+    res = env.execute_command('FT.SEARCH', 'idx', '*', 'sortby', 'numval_name', 'ASC',
+                            'RETURN', 8,'numval_name',
+                                        'numval_name', 'AS', 'numval_new_name',
+                                        'numval_name', 'AS', 'numval_new_name2',
+                                        'text_name')
+    unsorted_expected = ['key5', ['text_name', 'Meow'],
+                        'key3', [],
+                        'key4', []]
+    # First results should be the indexed documents that contains the numeric that determines the sorting order, 
+    # sorted by their value in ascending order
+    env.assertEqual(res[1:5], ['key2', ['numval_name', '109', 'numval_new_name', '109', 'numval_new_name2', '109'], 
+                             'key1', ['numval_name', '110', 'numval_new_name', '110', 'numval_new_name2', '110']])      
+    # Next, all other documents in the database, no order is guaranteed.
+    for val in unsorted_expected:
+        env.assertContains(val, res[5::])  
+
+    # If no `SORTBY', we expect the same results, different order.
+    # Because the first RETURN is the original path, the values are taken from redis and not from the 
+    # index.
+    res = env.execute_command('FT.SEARCH', 'idx', '*', 
+                              'RETURN', 4,'numval',
+                                          'numval_name', 'AS', 'numval_new_name')
+    env.assertEqual(res, [docs_num, 'key1', ['numval', '110', 'numval_new_name', '110'],
+                             'key2', ['numval', '109', 'numval_new_name', '109'], 
+                             'key3', [],
+                             'key4', [],
+                             'key5', []])  
+    
+    # `RETURN b as x
+    #         x as y` is allowed and yields: title = x, val = b title = y, val = x  
+    res = env.execute_command('FT.SEARCH', 'idx', '*', 
+                              'RETURN', 6,'numval_name','AS', 'x',
+                                          'x', 'AS', 'y')
+    env.assertEqual(res, [docs_num, 'key1', ['x', '110'],
+                             'key2', ['x', '109'], 
+                             'key3', [],
+                             'key4', ['y', '107'],
+                             'key5', []])
+    
+    # Test order of return - shouldn't change the result.
+    res2 = env.execute_command('FT.SEARCH', 'idx', '*', 
+                              'RETURN', 6,'x', 'AS', 'y',
+                                        'numval_name','AS', 'x')
+    env.assertEqual(res2, res)   
+
+def test_aliasing_sortables(env):
+    aliasing(env, is_sortable = True, is_sortable_unf = False)
+    
+def test_aliasing_NOTsortables(env):
+    
+    aliasing(env, is_sortable = False, is_sortable_unf = False)
+    
+def test_aliasing_sortables_UNF(env):
+    aliasing(env, is_sortable = True, is_sortable_unf = True)
+
```

Comment:
add `FT.PROFILE` output checks for the right result processors placements in multithreaded scenrios